DoorGate/DoorPresenter.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  DoorPresenter.swift
+//  
+//
+//  Created by Yuliani Noriega on 1/26/18.
+//
+import RxSwift
+
+enum DoorState :Int {
+    case empty = 0, occupied, locked
+}
+
+let THRESHOLD_TO_DISABLE_IN = 2
+let THRESHOLD_TO_DISABLE_OUT = 0
+
+struct DoorEntity {
+    var count :DoorState
+    var inEnable :Bool
+    var outEnable :Bool
+    
+    static func empty() -> DoorEntity {
+        return DoorEntity(count: .empty, inEnable: false, outEnable: false)
+    }
+}
+
+class DoorPresenter {
+    var entity = PublishSubject<DoorEntity>()
+    private let disposeBag = DisposeBag()
+
+    init(commands:PublishSubject<DoorCommand>) {
+        commands
+            .scan(DoorEntity.empty(), accumulator: DoorPresenter.accumulator)
+            .subscribe({ result in
+                self.entity.onNext(result.element!)
+        }).disposed(by: disposeBag)
+    }
+
+    static func accumulator(previousEntity: DoorEntity,
+                                    command: DoorCommand) -> DoorEntity {
+        let previousCount = previousEntity.count.rawValue
+        var newCount :DoorState
```

Comment:
Here I think it would be clearer to call the variable `newDoorState` instead of `newCount`